Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts

```diff
@@ -0,0 +1,387 @@
+            hdid: string
+        ): Promise<void> {
+            if (authenticatedVaccination.value.status === LoadStatus.LOADED) {
+                logger.debug(
+                    `Authenticated vaccination status found stored, not querying!`
+                );
+                return Promise.resolve();
+            } else {
+                logger.debug(`Retrieving authenticated vaccination status`);
+                setAuthenticatedRequested();
+                return vaccinationStatusService
+                    .getAuthenticatedVaccineStatus(hdid)
+                    .then((result) => {
+                        const payload = result.resourcePayload;
+                        if (result.resultStatus === ResultType.Success) {
+                            setAuthenticatedVaccinationStatus(payload);
+                        } else if (
+                            result.resultError?.actionCode ===
+                                ActionType.Refresh &&
+                            !payload.loaded &&
+                            payload.retryin > 0
+                        ) {
+                            logger.info(
+                                "Authenticated vaccination status not loaded"
+                            );
+                            setAuthenticatedVaccinationStatusMessage(
+                                "Please wait a moment while we retrieve your proof of vaccination."
+                            );
+                            setTimeout(() => {
+                                logger.info(
+                                    "Re-querying for authenticated proof of vaccination"
+                                );
+                                retrieveAuthenticatedVaccineStatus(hdid);
+                            }, payload.retryin);
+                        } else {
+                            if (result.resultError) {
+                                throw result.resultError;
+                            }
+                            logger.warn(
+                                `Authenticated vaccination status retrieval failed! ${JSON.stringify(
+                                    result
+                                )}`
+                            );
+                        }
+                    })
+                    .catch((error: ResultError) => {
+                        handleAuthenticatedError(error, ErrorType.Retrieve);
+                        throw error;
+                    });
+            }
+        }
+
+        function retrieveAuthenticatedVaccineRecord(
+            hdid: string
+        ): Promise<void> {
+            const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+            const vaccinationStatusService =
+                container.get<IVaccinationStatusService>(
+                    SERVICE_IDENTIFIER.VaccinationStatusService
+                );
+
+            logger.debug(`Retrieving authenticated vaccination record`);
```

Comment:
```suggestion
            logger.debug(`Retrieving authenticated vaccination record`);
```

These services have already been retrieved.